AI/VCAI/Goals.cpp

```diff
@@ -1125,13 +1131,13 @@ TGoalVec GatherArmy::getAllPossibleSubgoals()
 		}
 	}
 
-	if (ai->canRecruitAnyHero()) //this is not stupid in early phase of game
+	if(ai->canRecruitAnyHero()) //this is not stupid in early phase of game

```

Comment:
Since we not decided whatever canRecruitAnyHero should check available resources please move "ai->freeResources()[Res::GOLD] > GameConstants::HERO_GOLD_COST" check here for now.